docker/tarfile/src.go

```diff
@@ -74,7 +74,7 @@ func NewSourceFromFile(path string) (*Source, error) {
 func NewSourceFromStream(inputStream io.Reader) (*Source, error) {
 	// FIXME: use SystemContext here.
 	// Save inputStream to a temporary file
-	tarCopyFile, err := ioutil.TempFile(tmpdir.TemporaryDirectoryForBigFiles(), "docker-tar")
+	tarCopyFile, err := ioutil.TempFile(tmpdir.GetTemporaryDirectoryForBigFiles(), "docker-tar")
```

Comment:
This also needs to be enhanced to be overridable, by adding an extra parameter to `NewSourceFromStream`.

… except that it would be an API break.